interchain-token-service/src/proxy.rs

```diff
@@ -50,21 +49,92 @@ pub trait ProxyModule:
     + express_executor_tracker::ExpressExecutorTracker
     + multiversx_sc_modules::pause::PauseModule
 {
+    fn gas_service_pay_gas_for_contract_call(
+        &self,
+        destination_chain: &ManagedBuffer,
+        destination_address: &ManagedBuffer,
+        payload: &ManagedBuffer,
+        token_identifier: EgldOrEsdtTokenIdentifier,
+        gas_value: BigUint,
+    ) {
+        if token_identifier.is_egld() {
+            self.gas_service_pay_native_gas_for_contract_call(
```

Comment:
Again, no need for 2 different functions. You can do a check in the `pay_gas_for_contract_call` endpoint, to see if the payment is egld or an esdt.

Reply:
Check previous comment.